src/metadata.rs

```diff
@@ -76,6 +76,21 @@ fn path_to_content_type(path: &Path) -> String {
 }
 
 impl Metadata21 {
+    /// Merge metadata with pyproject.toml
+    ///
+    /// manifest_path must be the directory, not the file
+    pub fn merge_pyproject_toml(
+        &mut self,
+        pyproject_toml: &PyProjectToml,
+        manifest_path: impl AsRef<Path>,
+    ) -> Result<()> {
+        // FIXME: where to find `pyproject.toml`? relative to manifest_path? relative to working directory?
```

Comment:
I'm opening this draft PR for discussion.

@konstin  what's your thought about this?